src/util_cs/metric_cs.rs

```diff
@@ -45,16 +46,29 @@ impl Ord for OrderedVariable {
 pub struct MetricCS<Scalar: PrimeField> {
     named_objects: HashMap<String, NamedObject>,
     current_namespace: Vec<String>,
-    #[allow(clippy::type_complexity)]
-    constraints: Vec<(
-        LinearCombination<Scalar>,
-        LinearCombination<Scalar>,
-        LinearCombination<Scalar>,
-        String,
-    )>,
+    constraints: Vec<crate::util_cs::Constraint<Scalar>>,
     inputs: Vec<String>,
     aux: Vec<String>,
 }
+impl<Scalar: PrimeField> Comparable<Scalar> for MetricCS<Scalar> {
+    fn num_inputs(&self) -> usize {
+        self.num_inputs()
+    }
+    fn num_constraints(&self) -> usize {
+        self.num_constraints()
+    }
+
+    fn aux(&self) -> Vec<String> {
+        self.aux.clone()
+    }
+
+    fn inputs(&self) -> Vec<String> {
+        self.inputs.clone()
+    }
+    fn constraints(&self) -> Vec<crate::util_cs::Constraint<Scalar>> {
```

Comment:
this should return a slice, this is sometimes very large and expensive

Reply:
Done.